src/graphql.tsx

```diff
@@ -65,8 +65,33 @@ export interface GraphQLDataProps {
   updateQuery: (mapFn: (previousQueryResult: any, options: UpdateQueryOptions) => any) => void;
 }
 
-export interface InjectedGraphQLProps<T> {
-  data?: T & GraphQLDataProps;
+export type MutationFunc<TResult> = (opts: MutationOptions) => Promise<ApolloQueryResult<TResult>>;
+
+export interface OptionProps<TProps, TResult> {
+  ownProps: TProps;
+  data?: QueryProps & TResult;
+  mutate?: MutationFunc<TResult>;
+}
+
+export type DefaultChildProps<P, R> = P & { data?: QueryProps & R, mutate?: MutationFunc<R> };
+
+export interface OperationOption<TProps, TResult> {
+  options?: QueryOptions | MutationOptions | ((props: TProps) => QueryOptions | MutationOptions);
+  props?: (props: OptionProps<TProps, TResult>) => any;
+  skip?: boolean | ((props: any) => boolean);
+  name?: string;
+  withRef?: boolean;
+  shouldResubscribe?: (props: TProps, nextProps: TProps) => boolean;
```

Comment:
that should still be fine, they should only be optional if the call can pass the variables as `undefined`

```ts
interface Test {
  testFn (x: string, y: string): boolean;
}

let z: Test = { 
  // this is fine
  testFn: () => true
};
```